### PR TITLE
Update to RuboCop 0.36

### PR DESCRIPTION
**Pull request by tarcieri**
Includes a lot more cops for frozen string literals and constants which seem nice, particularly on Ruby 2.3+

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -61,17 +61,8 @@ Style/HashSyntax:
 Style/Lambda:
   Enabled: false
 
-Style/SingleSpaceBeforeFirstArg:
```
Comment (tarcieri) [inline .rubocop.yml]: This doesn't exist anymore?

File: lib/http/chainable.rb [hunk 2]
```diff
@@ -206,7 +205,7 @@ def basic_auth(opts)
       user = opts.fetch :user
       pass = opts.fetch :pass
 
-      auth("Basic " << Base64.strict_encode64("#{user}:#{pass}"))
+      auth("Basic " + Base64.strict_encode64("#{user}:#{pass}"))
```
Comment (tarcieri) [inline lib/http/chainable.rb]: "Basic" is now frozen here (see `# frozen_string_literal: true` at the top)

File: .rubocop.yml (hunk 1 above)
Comment (ixti) [inline .rubocop.yml]: It's now called: `Style/SpaceBeforeFirstArg`

**Comment by ixti** [comment]
:+1:

File: .rubocop.yml (hunk 1 above)
Comment (ixti) [inline .rubocop.yml]: In any case I believe it's fine to remove it...

**Pull request merged**